### ISSUE TITLE
Property 'className' does not exist on type 'IntrinsicAttributes'

### ISSUE THREAD

**Issue by hellhorse123** [pd4d10/vite-plugin-svgr#118]
In my React TS project with vite and `"vite-plugin-svgr": "^4.2.0"` I start getting error:

> Type '{ className: string; }' is not assignable to type 'IntrinsicAttributes'.
>   Property 'className' does not exist on type 'IntrinsicAttributes'.ts(2322)
> (property) className: string

when I trying to add className to svg:
`<Link className="goal-card__link_svg" />`

My vite.config.ts :
```
  plugins: [
    react(),
    svgr({
      include: "**/*.svg",
    }),
    ...
```

**Comment by xinzhang** [comment]
Yeah also if I want to do style={{}} or width={} height={}, will not work
is there a way to fix this?

**Comment by pd4d10** [comment]
You need to tweak this type definition file to keep consistent with the `include` option:

https://github.com/pd4d10/vite-plugin-svgr/blob/1280569bbaccc2fb7be43ba70f2b384f6cabba93/client.d.ts#L4

to `declare module "*.svg"`.

But I'm not sure if it would confilict with the Vite build-in types.

**Comment by cheshirrrcat** [comment]
Creating global.d.ts file and placing it in src folder helped me with resolve this issue

```typescript
declare module "*.svg" {
  import type * as React from "react";
  const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
  export default ReactComponent;
}
```

**Comment by hellhorse123** [comment]
> Creating global.d.ts file and placing it in src folder helped me with resolve this issue
> 
> ```ts
> declare module "*.svg" {
>   import type * as React from "react";
>   const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>;
>   export default ReactComponent;
> }
> ```

Yes, this way helps me

**Comment by jwalkerinterpres** [comment]
That way worked for me too, but the key thing was that I had to put it _before_ the line:

>/// <reference types="vite/client" />

If I put it after that declaration, it didn't work.

**Comment by AimanKM** [comment]
To me, this one works fine.
`declare module '*.svg' {
  import * as React from 'react'
  const ReactComponent: React.FC<React.SVGProps<SVGSVGElement>>
  export = ReactComponent
}`

<img width="623" alt="Image" src="https://github.com/user-attachments/assets/50ca6baa-1fca-4891-9b26-549e5f623c55" />

**Comment by AimanKM** [comment]
To me, this one works fine.

> declare module '*.svg' {
> import * as React from 'react'
> const ReactComponent: React.FC<React.SVGProps>
> export = ReactComponent
> }

<img width="779" alt="Image" src="https://github.com/user-attachments/assets/7f8bbebc-27d1-4bd5-a08d-e45fea8ebb3f" />
